code/modules/spacepods/spacepod.dm

```diff
@@ -272,61 +273,32 @@
 			to_chat(user, "\red The maintenance hatch is closed!")
 			return
 		if(!equipment_system)
-			to_chat(user, "<span class='warning'>The pod has no equipment datum, yell at pomf</span>")
+			to_chat(user, "<span class='warning'>The pod has no equipment datum, yell at IK3I</span>")
 			return
 		if(istype(W, /obj/item/device/spacepod_equipment/weaponry))
-			if(equipment_system.weapon_system)
-				to_chat(user, "<span class='notice'>The pod already has a weapon system, remove it first.</span>")
-				return
-			else
-				to_chat(user, "<span class='notice'>You insert \the [W] into the equipment system.</span>")
-				user.drop_item(W)
-				W.forceMove(src)
-				equipment_system.weapon_system = W
-				equipment_system.weapon_system.my_atom = src
-				return
-
+			add_equipment(user, W, "weapon_system")
+			return
 		if(istype(W, /obj/item/device/spacepod_equipment/misc))
-			if(equipment_system.misc_system)
-				to_chat(user, "<span class='notice'>The pod already has a miscellaneous system, remove it first.</span>")
-				return
-			else
-				to_chat(user, "<span class='notice'>You insert \the [W] into the equipment system.</span>")
-				user.drop_item(W)
-				W.forceMove(src)
-				equipment_system.misc_system = W
-				equipment_system.misc_system.my_atom = src
-				return
+			add_equipment(user, W, "misc_system")
+			return
 		if(istype(W, /obj/item/device/spacepod_equipment/cargo))
-			if(equipment_system.cargo_system)
-				to_chat(user, "<span class='notice'>The pod already has a cargo system, remove it first.</span>")
-				return
-			else
-				to_chat(user, "<span class='notice'>You insert \the [W] into the cargo system.</span>")
-				user.drop_item(W)
-				W.forceMove(src)
-				equipment_system.cargo_system = W
-				equipment_system.cargo_system.my_atom = src
-				return
+			add_equipment(user, W, "cargo_system")
+			return
+		if(istype(W, /obj/item/device/spacepod_equipment/sec_cargo))
+			add_equipment(user, W, "sec_cargo_system")
+			return
 		if(istype(W, /obj/item/device/spacepod_equipment/lock))
-			if(equipment_system.lock_system)
-				to_chat(user, "<span class='notice'>The pod already has a lock system, remove it first.</span>")
-				return
-			else
-				to_chat(user, "<span class='notice'>You insert \the [W] into the lock system.</span>")
-				user.drop_item(W)
-				W.forceMove(src)
-				equipment_system.lock_system = W
-				equipment_system.lock_system.my_atom = src
-				return
+			add_equipment(user, W, "lock_system")
+			return
 
 	if(istype(W, /obj/item/device/spacepod_key) && istype(equipment_system.lock_system, /obj/item/device/spacepod_equipment/lock/keyed))
 		var/obj/item/device/spacepod_key/key = W
 		if (key.id == equipment_system.lock_system.id)
-			locksecondseat()
+			lock_pod()
+			return
```

Comment:
You don't actually need a `return` statement here or in the `else` if you just leave `return` where it was before.